docs/_future/custom_functions.md

```diff
@@ -0,0 +1,146 @@
+* [call Slack API methods](/future/apicalls) or [third-party APIs](/future/apicalls/third-party)
+
+You can also encapsulate your business logic separately from the function handler, then import what you need and build your functions that way.
+
+#### 3. Add function listener and handler(s)
+Once your function has been defined, you'll need to register a function listener to trigger the function's functionality as well as any other related events.
+
+To do this, create a file in `listeners/functions`. This file will use the defined function and also implement any needed additional logic - in this case, it'll send a message to greet someone based on what is being passed in from the greeting function in step 2. 
+
+```js
+// For more information about functions: https://api.slack.com/future/functions
+const { SlackFunction } = require('@slack/bolt');
+
+// Get our Function Definition from the manifest!
+const { SampleFunctionDefinition } = require('../../manifest/function/sample-function');
+
+// Here is the work we want to do!
+const helloWorld = async ({ event, client, complete }) => {
+  const { recipient, channel, message } = event.inputs;
+  const salutations = ['Hello', 'Hi', 'Howdy', 'Hola', 'Salut'];
+  const salutation = salutations[Math.floor(Math.random() * salutations.length)];
+  try {
+    await client.chat.postMessage({
+      channel,
+      text: `${salutation}, <@${recipient}>! :wave: Someone sent the following greeting: \n\n>${message}`,
+    });
+    complete();
+  } catch (err) {
+    // Complete function with an error
+    await complete({ error: `There was an issue: ${err}` });
+    throw (err);
+  }
+};
+
+// Let's create a new Slack Function with helloWorld as its handler
+const helloWorldFunc = new SlackFunction(SampleFunctionDefinition.id, helloWorld);
+
+module.exports = { helloWorldFunc };
+```
+
+Once this file has been created, the function handler can be registered in an `index.js` file within the `listeners/functions` directory:
+```js
+const { helloWorldFunc } = require('./hello-world');
+
+// Register a complete function
+module.exports.register = (app) => {
+  app.function(helloWorldFunc);
+  // Register another function here
+};
+```
+
+In order to make sure this handler is triggered, make sure the Function listeners are registered in your `listeners/index.js` file:
+```js
+const functions = require('./functions');
+
+module.exports.registerListeners = (app) => {
+  functions.register(app);
+};
+```
+#### 4. Add function to app through workflow or manifest
+To actually call the defined function, `SampleFunctionDefinition`, don't forget to add your function to a workflow! When you're finished defining and implementing your functions, the next step is to add them to [Workflows](/bolt-js/future/workflows). Once added as a step in a Workflow, your Function will run when that Workflow is invoked by a [Trigger](/bolt-js/future/triggers).
```

Comment:
@srajiang I omitted the distribution of functions since it required deployment - let me know if you feel that we should add it back in based on what we decide to include in the deploy guide! (cc: @e-zim)